modules/xmpp/SdpConsistency.js

```diff
@@ -0,0 +1,175 @@
+ * @returns {number} the primary video ssrc
+ */
+function getPrimarySsrc (videoMLine) {
+    if (!videoMLine.ssrcs) {
+        return;
+    }
+    let numSsrcs = videoMLine.ssrcs
+        .map(ssrcInfo => ssrcInfo.id)
+        .filter((ssrc, index, array) => array.indexOf(ssrc) === index)
+        .length;
+    if (numSsrcs === 1) {
+        return videoMLine.ssrcs[0].id;
+    } else {
+        let findGroup = (mLine, groupName) => {
+            return mLine
+                .ssrcGroups
+                .filter(group => group.semantics === groupName)[0];
+        };
+        // Look for a SIM or FID group
+        if (videoMLine.ssrcGroups) {
+            let simGroup = findGroup(videoMLine, "SIM");
+            if (simGroup) {
+                return parseInt(simGroup.ssrcs.split(" ")[0]);
+            }
+            let fidGroup = findGroup(videoMLine, "FID");
+            if (fidGroup) {
+                return parseInt(fidGroup.ssrcs.split(" ")[0]);
+            }
+        }
+    }
+}
+
+/**
+ * Given a video mline (as parsed from transform.parse),
+ *  and a primary ssrc, return the corresponding rtx ssrc
+ *  (if there is one) for that video ssrc
+ * @param {object} videoMLine the video MLine from which to extract the
+ *  rtx video ssrc
+ * @param {number} primarySsrc the video ssrc for which to find the
+ *  corresponding rtx ssrc
+ * @returns {number} the rtx ssrc (or undefined if there isn't one)
+ */
+function getRtxSsrc (videoMLine, primarySsrc) {
+    if (videoMLine.ssrcGroups) {
+                let fidGroup = videoMLine.ssrcGroups.find(group => {
+            if (group.semantics === "FID") {
+              let groupPrimarySsrc = parseInt(group.ssrcs.split(" ")[0]);
+              return groupPrimarySsrc === primarySsrc;
+            }
+        });
+        if (fidGroup) {
+          return parseInt(fidGroup.ssrcs.split(" ")[1]);
+        }
+    }
+}
+/**
+ * End helper functions
+ */
+
+/**
+ * Handles the work of keeping video ssrcs consistent across multiple
+ * o/a cycles, making it such that all stream operations can be
+ * kept local and do not need to be signaled.
+ * NOTE: This only keeps the 'primary' video ssrcs consistent: meaning
+ * the primary video stream and an associated RTX stream, if it exists
+ */
+export default class SdpConsistency {
+    constructor () {
+        this.clearSsrcCache();
+    }
+
+    /**
+     * Clear the cached primary and primary rtx ssrcs so that
+     *  they will not be used for the next call to
+     *  makeVideoPrimarySsrcsConsistent
+     */
+    clearSsrcCache () {
+        this.cachedPrimarySsrc = null;
+        this.cachedPrimaryRtxSsrc = null;
+    }
+
+    /**
+     * Explicitly set the primary ssrc to be used in
+     *  makeVideoPrimarySsrcsConsistent
+     * @param {number} primarySsrc the primarySsrc to be used
+     *  in future calls to makeVideoPrimarySsrcsConsistent
+     */
+    setPrimarySsrc (primarySsrc) {
+        this.cachedPrimarySsrc = primarySsrc;
```

Comment:
yup...the division of responsibilities is:
SdpConsistency keep the primary video ssrc consistent
SdpSimulcast adds simulcast ssrcs and takes the responsibility for keeping the sim ssrcs consistent
(And, in a future PR: RtxModifier adds rtx ssrcs and keeps them consistent)